logic/scripts/update_token_insight.py

```diff
@@ -366,16 +398,221 @@ def fetch_wallet_balance(wallet, label, eth_threshold):
+    team_dist_balance = ogn_balances[team_dist_address]
+    investor_dist_balance = ogn_balances[investor_dist_address]
+    dist_staging_balance = ogn_balances[dist_staging_address]
+    partnerships_balance = ogn_balances[partnerships_address]
+    ecosystem_growth_balance = ogn_balances[ecosystem_growth_address]
+
+    reserved_tokens = int(
+        foundation_reserve_balance +
+        team_dist_balance +
+        investor_dist_balance +
+        dist_staging_balance +
+        partnerships_balance +
+        ecosystem_growth_balance
+    )
+
+    circulating_supply = int(total_supply - reserved_tokens)
+
+    market_cap = int(circulating_supply * ogn_usd_price)
+
+    out_data = dict([
+        ("ogn_usd_price", ogn_usd_price),
+        ("circulating_supply", circulating_supply),
+        ("market_cap", market_cap),
+        ("total_supply", total_supply),
+
+        ("reserved_tokens", reserved_tokens),
+        ("staked_user_count", staked_user_count),
+        ("staked_token_count", staked_token_count),
+
+        ("foundation_reserve_address", foundation_reserve_address),
+        ("team_dist_address", team_dist_address),
+        ("investor_dist_address", investor_dist_address),
+        ("dist_staging_address", dist_staging_address),
+        ("partnerships_address", partnerships_address),
+        ("ecosystem_growth_address", ecosystem_growth_address),
+    ])
+
+    if format_data:
+        out_data["ogn_usd_price"] = '${:,}'.format(ogn_usd_price)
+        out_data["circulating_supply"] = '{:,}'.format(circulating_supply)
+        out_data["market_cap"] = '{:,}'.format(market_cap)
+        out_data["total_supply"] = '{:,}'.format(total_supply)
+        out_data["reserved_tokens"] = '{:,}'.format(reserved_tokens)
+        out_data["staked_user_count"] = '{:,}'.format(staked_user_count)
+        out_data["staked_token_count"] = '{:,}'.format(staked_token_count)
+
+    return out_data
+    
+def get_supply_history():
+    data =  redis_client.get("ogn_supply_data") or "[]"
+
+    return data
+
+def update_circulating_supply():
+    stats = get_ogn_stats(format_data=False)
+    snapshot_date = datetime.utcnow()
+
+    supply_snapshot = db_common.get_or_create(
+        db.session, db_models.CirculatingSupply, snapshot_date=snapshot_date
+    )
+
+    supply_snapshot.supply_amount = stats["circulating_supply"]
+    db.session.commit()
+
+    supply_data = db.engine.execute("""
+    select timewin, max(s.supply_amount)
+    from 
+        generate_series(now() - interval '12 month', now(), '1 day') as timewin
+    left outer join 
+        (select * from circulating_supply where snapshot_date > now() - interval '12 month' and snapshot_date > '2020-01-01'::date order by snapshot_date desc) s
+    on s.snapshot_date < timewin 
+        and s.snapshot_date >= timewin - (interval '1 day')
+    where timewin > '2020-01-01'::date
+    group by timewin
+    order by timewin desc
+    """)
```

Comment:
@franckc this is the query

Reply:
@shahthepro Is there an index on circulating_supply.snapshot_date? It might help speed up the query. Otherwise I'd recommend profiling the query using "EXPLAIN" in postgres. See https://www.postgresql.org/docs/9.3/sql-explain.html

Reply:
@franckc Yes there is an index on that column. Will try to profile it.